scripts/check-yaml.py

```diff
@@ -49,8 +51,41 @@ def check(self) -> int:
                         "The \"%s\" file must be non-empty",
                         taxonomy.path.with_name(attribution_path.name),
                     )
+                # NOTE: The following three warnings are intended for the
+                # beta only, at the moment, and only to flag issues for
+                # maintainers to address rather than block on them. We will
+                # revisit when other content is allowed.
+                qna_file_path = taxonomy.rel_path.with_name("qna.yaml")
+                if "knowledge" in qna_file_path.parts:
+                    qna_file_contents = parser.parse(qna_file_path).contents
+                    for element in qna_file_contents["document"]["patterns"]:
+                        if not re.match('.*.md', element):
+                            taxonomy.warning(
+                                "The document \"%s\" should be a markdown "
+                                "file.",
+                                element
+                            )
+                    if not re.match(
+                            'https://github\.com\/.*',
+                            qna_file_contents["document"]["repo"]):
+                        taxonomy.warning(
+                            "The document repo \"%s\" needs to be a "
+                            "GitHub-based repository.",
+                            qna_file_contents["document"]["repo"]
+                        )
+                    if not re.match(
```

Comment:
This potential check is discussed here: https://github.com/instructlab/schema/issues/30

If we do want to require SHA values, we should probably do that in the schema. But I am not convinced that is a great idea. Since we could allow non-SHA values that have special meanings.